web/controllers/package_controller.ex

```diff
@@ -3,24 +3,43 @@ defmodule HexWeb.PackageController do
 
   @packages_per_page 30
   @sort_params ~w(name downloads inserted_at)
+  @letters for letter <- ?A..?Z, do: <<letter>>
 
   def index(conn, params) do
+    letter        = HexWeb.Utils.safe_search(params["letter"])
     search        = HexWeb.Utils.safe_search(params["search"])
+    maintainer    = params["maintainer"]
+
+    filter =
+      cond do
+        letter ->
+          {:starts_with, letter}
+        maintainer ->
+          {:maintainer, maintainer}
```

Comment:
Sure, I realise that it's just a text field, but I think it's still fine to include as a feature,

* People will generally be entering their own name when they are a maintainer, thus spelling mistakes are unlikely
* There is little harm if a name is misspelled, we are just potentially not showing one of their packages in the list
* If this happens people are likely to fix names with errors 

IMO the benefit outweighs the low downside.